mrp_subcontracting_purchase_link/models/mrp_production.py

```diff
@@ -8,5 +8,13 @@ class MrpProduction(models.Model):
     _inherit = "mrp.production"
 
     purchase_order_id = fields.Many2one(
-        "purchase.order", "Subcontract Purchase Order", readonly=True
+        "purchase.order",
+        "Subcontract Purchase Order",
+        readonly=True,
+        related="purchase_line_id.order_id",
+        store=True,
+    )
+
+    purchase_line_id = fields.Many2one(
```

Comment:
Another approach could be to make this a stored compute field with a compute method like below, removing the extension of `_prepare_subcontract_mo_vals()`.

```
@api.depends("move_dest_ids")
def _compute_purchase_line_id(self):
    for production in self:
        if production.move_dest_ids:
            production.purchase_line_id = production.move_dest_ids[0].purchase_line_id
```

If we take this approach, though, we may need to prepare a pre-init hook script for newly installing this module in an existing database with many subcontracting transactions.

Reply:
In the case of subcontracting, it makes sense to have the purchase order line in the manufacturing order, as it provides coherence when passed in _prepare_subcontract_mo_vals because you have it directly